base/config.js

```diff
@@ -22,18 +22,26 @@ module.exports.schema = {
     message: 'should be a function or array of functions',
     validate: value => typeof value === 'function' || (isArray(value) && filter(value, f => typeof f === 'function').length === value.length)
   },
-  endpoint: {
-    defaultValue: () => 'https://notify.bugsnag.com',
-    message: 'should be a URL',
-    validate: () => true
-  },
-  sessionEndpoint: {
-    defaultValue: () => 'https://sessions.bugsnag.com',
-    message: 'should be a URL',
-    validate: () => true
+  endpoints: {
+    defaultValue: () => ({
+      notify: 'https://notify.bugsnag.com',
+      sessions: 'https://sessions.bugsnag.com'
+    }),
+    message: 'should be an object containing endpoint URLs { notify, sessions }. sessions is optional if autoCaptureSessions=false',
+    validate: (val, obj) =>
+      // first, ensure it's an object
+      (val && typeof val === 'object') &&
+      (
+        // endpoints.notify must always be set
+        stringWithLength(val.notify) &&
+        // endpoints.sessions must be set unless session tracking is explicitly off
+        (obj.autoCaptureSessions === false || stringWithLength(val.sessions))
+      ) &&
+      // ensure no keys other than notify/session are set on endpoints object
+      filter(keys(val), k => !includes([ 'notify', 'sessions' ], k)).length === 0
   },
   autoCaptureSessions: {
-    defaultValue: () => false,
+    defaultValue: (val, opts) => opts.endpoints === undefined || (!!opts.endpoints && !!opts.endpoints.sessions),
```

Comment:
This is just my curious question about JS syntax - what does `!!` do?

Reply:
`!` is the logical _NOT_ operator. `!!` is just the logical _NOT_ operator repeated! 

Because js has the concept of "truthyness", you can use the _NOT_ operator twice on a truthy value to get the value `true`.

for example:

```js
var str = 'hi jamie'
var strExists = !!str //=> strExists = true

var nostr = null
var nostrExists = !!nostr //=> nostrExists = false
```

Why this is useful in this situation… the logical _AND_ operator `&&` will return the result of the right hand side if all the value are truthy:

```js
var opts = { endpoints: { sessions: 'yes' } }
var result = opts.endpoints && opts.endpoints.sessions
//=> result = 'yes'
```

We can use the `!` operator once to coerce the values into booleans, but then it is `false`, so we use it again to flip it back to `true`.

```js
var opts = { endpoints: { sessions: 'yes' } }
var result = !!opts.endpoints && !!opts.endpoints.sessions
//=> result = true
```

Hope this makes sense!